### PR TITLE
feat: problem error processor

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

**Comment by sdelamo** [comment]
@ilopmar could you review this ?

File: problem-json/src/test/groovy/io/micronaut/problem/ConstraintViolationProblemSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+        when:
+        Optional<Map> bodyOptional = e.response.getBody(errorArg)
+
+        then:
+        bodyOptional.isPresent()
+        bodyOptional.get().keySet().size() == 4
+        bodyOptional.get()['status'] == 400
+        bodyOptional.get()['title'] == 'Constraint Violation'
+        bodyOptional.get()['type'] == "https://zalando.github.io/problem/constraint-violation"
```
Comment (ilopmar) [inline problem-json/src/test/groovy/io/micronaut/problem/ConstraintViolationProblemSpec.groovy:50]: Can you also assert the content of the `violations` field?

File: problem-json/src/main/java/io/micronaut/problem/package-info.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * limitations under the License.
+ */
+/**
+ * Contains classes for rendering error as Problem in a Micronaut Application.
+ * @see <a href="https://tools.ietf.org/html/rfc7807">RFC 7807: Problem Details for HTTP APIs</a>
+ * @author Sergio del Amo
+ * @since 1.0
+ */
+@Configuration
+@Requires(property = ProblemConfigurationProperties.PREFIX + ".enabled", notEquals = StringUtils.FALSE)
```
Comment (ilopmar) [inline problem-json/src/main/java/io/micronaut/problem/package-info.java:23]: Add `defaultValue = StringUtils.TRUE`

File: settings.gradle [hunk 3]
```diff
@@ -1,3 +1,6 @@
-rootProject.name = 'xxx'
```
Comment (ilopmar) [inline settings.gradle:1]: When you rename this you didn't remove the `xxx` directory.

**Review by ilopmar — CHANGES_REQUESTED** [review]
Just some minor comments. Everything looks really good! :100:

File: settings.gradle (hunk 3 above)
Comment (sdelamo) [reply settings.gradle:1]: Thanks, I have deleted it.

File: problem-json/src/main/java/io/micronaut/problem/package-info.java (hunk 2 above)
Comment (sdelamo) [reply problem-json/src/main/java/io/micronaut/problem/package-info.java]: Done

File: problem-json/src/test/groovy/io/micronaut/problem/ConstraintViolationProblemSpec.groovy (hunk 1 above)
Comment (sdelamo) [reply problem-json/src/test/groovy/io/micronaut/problem/ConstraintViolationProblemSpec.groovy:50]: Done

**Review by ilopmar — APPROVED** [review]

**Pull request merged**